packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts

```diff
@@ -131,17 +134,53 @@ export class AuthService {
     );
   }
 
+  private async canUserBypassAuthProvider({
+    user,
+    workspace,
+    provider,
+  }: {
+    user: UserEntity;
+    workspace: WorkspaceEntity;
+    provider: AuthProviderEnum;
+  }): Promise<boolean> {
+    const isBypassEnabledForProvider = (() => {
+      switch (provider) {
+        case AuthProviderEnum.Password:
+          return workspace.isPasswordAuthBypassEnabled;
+        case AuthProviderEnum.Google:
+          return workspace.isGoogleAuthBypassEnabled;
+        case AuthProviderEnum.Microsoft:
+          return workspace.isMicrosoftAuthBypassEnabled;
+        default:
+          return false;
+      }
+    })();
+
+    if (!isBypassEnabledForProvider) {
+      return false;
+    }
+
+    const userWorkspace =
+      await this.userWorkspaceService.checkUserWorkspaceExists(
+        user.id,
+        workspace.id,
+      );
+
+    if (!userWorkspace) {
+      return false;
+    }
+
+    return await this.permissionsService.userHasWorkspaceSettingPermission({
+      userWorkspaceId: userWorkspace.id,
+      workspaceId: workspace.id,
+      setting: PermissionFlagType.SSO_BYPASS,
+    });
+  }
+
   async validateLoginWithPassword(
     input: UserCredentialsInput,
     targetWorkspace?: WorkspaceEntity,
   ) {
-    if (targetWorkspace && !targetWorkspace.isPasswordAuthEnabled) {
-      throw new AuthException(
-        'Email/Password auth is not enabled for this workspace',
-        AuthExceptionCode.FORBIDDEN_EXCEPTION,
-      );
-    }
-
     const user = await this.userRepository.findOne({
```

Comment:
By moving the condition check below you're probably allowing an attacker to know if a user exists or not on an SSO-only workspace which can be considered a security issue